### PR TITLE
fix(elasticsearch): Defining 2 domains with logging enabled in the same stack fails on construct id conflict

### LINKED ISSUE #12017: (elasticsearch): Two ES Domains with logging cannot be created in the same stack
When 2 ES Domains, both with logging enabled, are created in the same stack, a Construct naming conflict occurs

Error:
> There is already a Construct with name 'AppLogs' in Stack [es-bug-stack]

### Reproduction Steps

```
#!/usr/bin/env node
import { App, Stack } from '@aws-cdk/core';
import { Domain, ElasticsearchVersion } from '@aws-cdk/aws-elasticsearch';

const app = new App();
const stack = new Stack(app, 'es-bug-stack');
new Domain(stack, 'domain1', {
    version: ElasticsearchVersion.V7_7,
    logging: { appLogEnabled: true }
})
new Domain(stack, 'domain2', {
    version: ElasticsearchVersion.V7_7,
    logging: { appLogEnabled: true }
})
app.synth()
```

```
$ cdk synth
There is already a Construct with name 'AppLogs' in Stack [es-bug-stack]
Subprocess exited with error 1
```

### What did you expect to happen?

That 2 ES Domains can co-exist in the same stack

### What actually happened?

`cdk synth` throws the error:
> There is already a Construct with name 'AppLogs' in Stack [es-bug-stack]

### Environment

  - **CDK CLI Version  : 1.70.0
  - **Framework Version: 1.70.0
  - **Node.js Version:** v12.18.4
  - **OS               : macos
  - **Language (Version):** all

### Other

https://github.com/aws/aws-cdk/blob/ae2e9c1a40070509e2fb5a95de8881906b500546/packages/%40aws-cdk/aws-elasticsearch/lib/domain.ts#L1399

The following constructs have statically named construct LogicalIds: `SlowSearchLogs`, `SlowIndexLogs` & `AppLogs` which creates a conflict when 2 Domains exist in the same stack

```
    // Setup logging
    const logGroups: logs.ILogGroup[] = [];

    if (props.logging?.slowSearchLogEnabled) {
      this.slowSearchLogGroup = props.logging.slowSearchLogGroup ??
        new logs.LogGroup(scope, 'SlowSearchLogs', {
          retention: logs.RetentionDays.ONE_MONTH,
        });

      logGroups.push(this.slowSearchLogGroup);
    };

    if (props.logging?.slowIndexLogEnabled) {
      this.slowIndexLogGroup = props.logging.slowIndexLogGroup ??
        new logs.LogGroup(scope, 'SlowIndexLogs', {
          retention: logs.RetentionDays.ONE_MONTH,
        });

      logGroups.push(this.slowIndexLogGroup);
    };

    if (props.logging?.appLogEnabled) {
      this.appLogGroup = props.logging.appLogGroup ??
        new logs.LogGroup(scope, 'AppLogs', {
          retention: logs.RetentionDays.ONE_MONTH,
        });

      logGroups.push(this.appLogGroup);
    };
```

--- 

This is :bug: Bug Report

### PR DESCRIPTION
**Pull request by peterb154**
2 elastic search Domains with logging enabled and deployed in the same stack fail to synth due to log group logical naming conflict.

BREAKING CHANGE: ES Domain LogGroup LogicalId will change, which will trigger new log group resources to be created

Fixes #12017

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-elasticsearch/test/domain.test.ts [hunk 1]
```diff
@@ -81,7 +83,7 @@ describe('log groups', () => {
         SEARCH_SLOW_LOGS: {
           CloudWatchLogsLogGroupArn: {
             'Fn::GetAtt': [
-              'SlowSearchLogsE00DC2E7',
+              cdkAssert.stringLike(`${logicalId}SlowSearchLogs*`),
```
Comment (iliapolo) [inline packages/@aws-cdk/aws-elasticsearch/test/domain.test.ts:86]: Assert the concrete logical id that contains the domain as well. (in all tests)

**Review by iliapolo — DISMISSED** [review]

File: packages/@aws-cdk/aws-elasticsearch/test/domain.test.ts (hunk 1 above)
Comment (peterb154) [reply packages/@aws-cdk/aws-elasticsearch/test/domain.test.ts:86]: @iliapolo - so we should not use matchers for things like this? I assume that we *want* the test to break if the logicaIId changes?

Comment (iliapolo) [reply packages/@aws-cdk/aws-elasticsearch/test/domain.test.ts:86]: @peterb154 exactly

Comment (peterb154) [reply packages/@aws-cdk/aws-elasticsearch/test/domain.test.ts]: @iliapolo amended the commit with concrete logicalIds

**Review by iliapolo — APPROVED** [review]
Awesome, thanks for this :)

**Pull request merged**